#!/usr/bin/env python
PACKAGE="image_manip"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("frame_rate", double_t, 1, "frame rate Hz, 0 is maximum", 20, 0, 100)
# TODO need coarse and fine adjustment controls for these?
gen.add("offset", double_t, 1, "offset", 0, -256.0, 256.0)
gen.add("scale", double_t, 1, "scale", 1.0, -256.0, 256.0)

mode_enum = gen.enum([
  gen.const("passthrough", int_t, -1, "passthrough"),
  gen.const("8UC", int_t, 0, "8UC"),
  gen.const("8SC", int_t, 1, "8SC"),
  gen.const("16UC", int_t, 2, "16UC"),
  gen.const("16SC", int_t, 3, "8SC"),
  gen.const("32SC", int_t, 4, "32SC"),
  gen.const("32FC", int_t, 5, "32FC"),
  gen.const("64FC", int_t, 6, "64FC"),
  # gen.const("rbg8", int_t, 0, "rgb8"),
  # gen.const("rbga8", int_t, 1, "rgba8"),
  # gen.const("rbg16", int_t, 1, "rgb16"),
  # gen.const("rbga16", int_t, 1, "rgba16"),
  # gen.const("bgr8", int_t, 1, "bgr8"),
  # gen.const("bgra8", int_t, 1, "bgra8"),
  # gen.const("bgr16", int_t, 1, "bgr16"),
  # gen.const("bgra16", int_t, 1, "bgra16"),
  # gen.const("mono8", int_t, 1, "mono8"),
  # gen.const("mono16", int_t, 1, "mono16"),
  # No mono + alpha?
  ], "convert mode")
gen.add("dst_type", int_t, 0, "destination type", -1, -1, 6, edit_method=mode_enum)

exit(gen.generate(PACKAGE, "image_manip", "Convert"))
